fibo/dev_classes/phase4_UI/jira_pages.py

```diff
@@ -0,0 +1,192 @@
+    _LOGOUT_BUTTON = (By.ID, "log_out")
+    _FILTER_BASIC_A = (By.CSS_SELECTOR, "[data-id='advanced']")
+
+    def __init__(s, webdr):
+        super().__init__(webdr)
+
+    def open_filter(s):
+        s.tillInvisible(s._LOADING)
+        s.elem(s._ISSUES_MENU_LINK).click()
+        s.elem(s._SEARCHFOR_SUBMENU_LINK).click()
+        s.elem(s._ISSUE_HEADER)
+        s.tillInvisible(s._LOADING)
+
+    def open_reported_by_me(s):
+        if not s.ispresent(s._ISSUE_HEADER):
+            s.open_filter()
+        s.elem(s._REPORTED_BY_ME_LINK).click()
+        s.elem(s._REPORTED_HEADER)
+        s.tillInvisible(s._LOADING)
+
+    def create_issue(s, proj, type, sum):
+        s.elem(s._CREATE_LINK).click()
+        s.elem(s._PROJECT_INPUT).click()
+        s.elem(s._PROJECT_INPUT).send_keys(proj)
+        #
+        s.tillInvisible(s._EXPANDED_HEADER)
+        s.elem(s._ISSUE_TYPE_INPUT).click()
+        s.elem(s._ISSUE_TYPE_INPUT).send_keys(type + "\n")
+        #
+        s.tillInvisible(s._EXPANDED_HEADER)
+        s.elem(s._SUMMARY_HEADER).clear()
+        s.elem(s._SUMMARY_HEADER).send_keys(sum)
+        s.elem(s._CREATE_ISSUE_BUTTON).click()
+        if 0 < len(sum) < 255:
+            s.tillInvisible(s._CONFIRM_BUTTON)
+
+    def cancel_creation(s):
+        s.elem(s._CANCEL_BUTTON).click()
+        s.wd.switch_to.alert.accept()
+        s.tillInvisible(s._CONFIRM_BUTTON)
+
+    def select_issue(s, sum):
+        s.tillInvisible(s._LOADING)
+        s.elem((By.XPATH, f"//span[.='{sum}']")).click()
+        s.tillInvisible(s._LOADING)
+
+    def delete_issue(s):
+        s.elem(s._MORE_BUTTON).click()
+        s.elem(s._DEL_BUTTON).click()
+        s.elem(s._DEL_CONFIRM_BUTTON).click()
+        s.tillInvisible(s._LOADING)
+        s.tillInvisible(s._DELETE_MESSAGE)
+
+    def search_issue(s, sum):
+        if (s.isvisible(s._FILTER_BASIC_A)):
+            s.elem(s._FILTER_BASIC_A).click()
+        s.elem(s._SEARCHER_QUERY_INPUT).clear()
+        s.elem(s._SEARCHER_QUERY_INPUT).send_keys(sum + "\n")
+        s.tillInvisible(s._LOADING)
+
+    def count_filtered_issues(s):
+        return s.countelem(s._FILTERED_LI)
+
+    def trigger_update(s):
+        s.elem(s._UPDATE_BUTTON).click()
+        s.elem(s._SUMMARY_HEADER)
+
+    def confirm_update(s):
+        s.elem(s._UPDATE_CONFIRM_BUTTON).click()
+        s.tillInvisible(s._LOADING)
+        s.tillInvisible(s._UPDATE_MESSAGE)
+
+    def update_summary(s, new_sum):
+        if not s.ispresent(s._SUMMARY_HEADER):
+            s.trigger_update()
+        s.elem(s._SUMMARY_HEADER).clear()
+        s.elem(s._SUMMARY_HEADER).send_keys(new_sum + "\n")
+        s.tillInvisible(s._UPDATE_MESSAGE)
+
+    def update_priority(s, new_priority):
+        if not s.ispresent(s._SUMMARY_HEADER):
+            s.trigger_update()
+        s.elem(s._PRIORITY_TYPE_INPUT).click()
+        s.elem(s._PRIORITY_TYPE_INPUT).send_keys(new_priority + "\n")
+
+    def assign_to_me(s):
```

Comment:
It would be better to write click_on_assign_to_me_button.
Your colleague may not know how exactly you are going to assign the ticket to you.